### PR TITLE
[NEURON][TF][BUILD] - Neuron SDK 2.5.0 release

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [ ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [ ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 import test.test_utils as test_utils
 
 
-@pytest.mark.model("mnist")
+@pytest.mark.model("simple")
```
Comment (xncqr) [inline test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py:12]: Is there a reason for straying away from a common test model of mnist?

File: test/dlc_tests/sanity/test_telemetry.py [hunk 2]
```diff
@@ -392,7 +392,10 @@ def get_tensorflow_inference_command_tf27_above(image_uri, model_name):
 
     _, image_framework_version = test_utils.get_framework_and_version_from_tag(image_uri)
     if Version(image_framework_version) in SpecifierSet(">=2.7"):
-        inference_command = test_utils.build_tensorflow_inference_command_tf27_and_above(model_name)
+        entrypoint = "/usr/bin/tf_serving_entrypoint.sh"
+        if "neuron" in image_uri:
+            entrypoint = "/usr/local/bin/entrypoint.sh"
```
Comment (xncqr) [inline test/dlc_tests/sanity/test_telemetry.py:397]: Wondering why we are moving this, instead of keeping the original "copied" name within the docker image.

File: test/test_utils/__init__.py [hunk 3]
```diff
@@ -1774,13 +1779,13 @@ def get_tensorflow_model_base_path(image_uri):
     return model_base_path
 
 
-def build_tensorflow_inference_command_tf27_and_above(model_name):
+def build_tensorflow_inference_command_tf27_and_above(model_name, entrypoint="/usr/bin/tf_serving_entrypoint.sh"):
```
Comment (xncqr) [inline test/test_utils/__init__.py:1782]: This goes along the comment in test_telemetry. Youare adding even more logic to handle a file location/name situation where all that can be handled in the dockerfile on the "COPY" line, making what file you are copying, use the destination path/name which is already standard for the DLC Framework.

File: test/test_utils/ecs.py [hunk 4]
```diff
@@ -807,7 +807,10 @@ def setup_ecs_inference_service(
         model_base_path = get_tensorflow_model_base_path(docker_image_uri)
         _, image_framework_version = get_framework_and_version_from_tag(docker_image_uri)
         if Version(image_framework_version) in SpecifierSet(">=2.7"):
-            arguments_dict["container_command"] = [build_tensorflow_inference_command_tf27_and_above(model_name)]
+            entrypoint = "/usr/bin/tf_serving_entrypoint.sh"
+            if processor == "neuron":
+                entrypoint = "/usr/local/bin/entrypoint.sh"
+            arguments_dict["container_command"] = [build_tensorflow_inference_command_tf27_and_above(model_name, entrypoint)]
```
Comment (xncqr) [inline test/test_utils/ecs.py:813]: This is the same as the last two comments as it seems we are adding more logic where the main issue can be solved within the dockerfile.

File: test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py (hunk 1 above)
Comment (aws-vrnatham) [reply test/dlc_tests/eks/tensorflow/inference/test_eks_tensorflow_inference.py:12]: Oh ok.. This change is not needed. The model in the test is still mnist based. Will change the above back to mnist.

File: test/dlc_tests/sanity/test_telemetry.py (hunk 2 above)
Comment (aws-vrnatham) [reply test/dlc_tests/sanity/test_telemetry.py:397]: From the first release of neuron dlc for tf we have been using entrypoint.sh as the script (instead of tf_serving_entrypoint.sh). Documentation for say using DLC on inferentia [ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs-inference.html) have reference to the entrypoint.sh and hence using that in the test code also to make sure that it gets tested. I could copy the entrypoint.sh  to tf_serving_entrypoint.sh in dockerfile and not modify this test, but that could mean entrypoint.sh never gets tested and that is what the customer could be using.

File: test/test_utils/__init__.py (hunk 3 above)
Comment (aws-vrnatham) [reply test/test_utils/__init__.py:1782]: Same as above.

File: test/test_utils/ecs.py (hunk 4 above)
Comment (aws-vrnatham) [reply test/test_utils/ecs.py:813]: Yes. I understand but neuron DLC's have always been using entrypoint.sh.

Comment (xncqr) [reply test/test_utils/ecs.py:813]: If this is the case, why is this being added now, versus earlier, if this has always been done?

File: test/dlc_tests/sanity/test_telemetry.py (hunk 2 above)
Comment (aws-vrnatham) [reply test/dlc_tests/sanity/test_telemetry.py:397]: Earlier release was for TF2.5 and this new code takes into effect only for tf27 and above and that did not have the neuron changes.

File: test/test_utils/ecs.py (hunk 4 above)
Comment (aws-vrnatham) [reply test/test_utils/ecs.py:813]: The code added was specific to tf27 and above and the last neuron DLC release was for tf25.

File: tensorflow/inference/docker/2.8/py3/sdk2.5.0/Dockerfile.neuron.os_scan_allowlist.json [hunk 5]
```diff
@@ -0,0 +1,2560 @@
...
+            "cvss_v2_score": 0,
+            "cvss_v3_severity": "HIGH",
+            "source_url": "https://nvd.nist.gov/vuln/detail/CVE-2022-36003",
+            "source": "NVD",
+            "severity": "HIGH",
+            "status": "ACTIVE",
+            "title": "CVE-2022-36003 - tensorflow"
+        }
+    ]
+}
```
Comment (xncqr) [inline tensorflow/inference/docker/2.8/py3/sdk2.5.0/Dockerfile.neuron.os_scan_allowlist.json:2560]: Can we add a new line here to handle best practices. Note the red circle in the PR at the bottom of this file.

**Review by xncqr — APPROVED** [review]
LGTM. Previous comments resolved.
